context/src/main/java/io/opentelemetry/context/propagation/TextMapGetter.java

```diff
@@ -33,4 +35,20 @@ public interface TextMapGetter<C> {
    */
   @Nullable
   String get(@Nullable C carrier, String key);
+
+  /**
+   * If implemented, returns all values for a given {@code key} in order, or returns an empty list.
+   *
+   * <p>The default method returns the first value of the given propagation {@code key} as a
+   * singleton list, or returns an empty list.
+   *
+   * @param carrier carrier of propagation fields, such as an http request.
+   * @param key the key of the field.
+   * @return the first value of the given propagation {@code key} as a singleton list, or returns an
+   *     empty list.
+   */
+  default List<String> getList(@Nullable C carrier, String key) {
```

Comment:
Perhaps using `List<String>` isn't the best option. Maybe `Iterable<String>` as used in `keys` or `Iterator<String>` would work better? When using `List<String>` and carrier is `HttpServletRequest` where `getHeaders` returns `Enumeration` you'd have to copy the enumeration to list with `Collections.list(carrier.getHeaders(key))`. When using `Iterator<String>` you could skip copying the enumeration to list with
```java
Enumeration<String> enumeration = carrier.getHeaders(key);
return new Iterator<String>() {
  @Override
  public boolean hasNext() {
    return enumeration.hasMoreElements();
  }

  @Override
  public String next() {
    return enumeration.nextElement();
  }
};
```

Reply:
Yes I agree `Iterator<String>` is a better choice given the use of `Enumeration` in `HttpServletRequest`. I had just copied @jack-berg's implementation from #6837. Will change it once we get alignment with Jack.

Reply:
Hmm... iterator is easier / less allocation for some `TextMapGetter` implementations, but (slighty) worse ergonomics for callers. 

Given this, I think iterator is the better choice since we expect most (all?) propagator instances to be maintained by opentelemetry, and its fine for us to have slightly more burden to improve performance.

Reply:
Cool, I've changed it to `Iterator`. I've also renamed the method to `getAll()`